executor/aggregate.go

```diff
@@ -174,30 +172,21 @@ type HashAggInput struct {
 
 // HashAggIntermData indicates the intermediate data of aggregation execution.
 type HashAggIntermData struct {
-	groupKeys   [][]byte
-	cursor      int
-	groupCtxMap aggCtxsMapper
+	groupKeys        [][]byte
+	cursor           int
+	partialResultMap aggPartialResultMapper
 }
 
-// ToRows converts HashAggInterData to DatumRows.
-func (d *HashAggIntermData) ToRows(sc *stmtctx.StatementContext, rows [][]types.Datum, aggFuncs []aggregation.Aggregation, maxChunkSize int) (_ [][]types.Datum, reachEnd bool) {
-	if len(rows) == maxChunkSize {
-		return rows, false
+// getPartialResultBatch fetches a batch of partial results from HashAggIntermData.
+func (d *HashAggIntermData) getPartialResultBatch(sc *stmtctx.StatementContext, prs [][]aggfuncs.PartialResult, aggFuncs []aggfuncs.AggFunc, maxChunkSize int) (_ [][]aggfuncs.PartialResult, groupKeys [][]byte, reachEnd bool) {
+	if len(prs) == maxChunkSize {
```

Comment:
It seems that this function returned all the `(group key, partial result)` pairs back to the caller, this check can be removed?